src/main/java/org/apache/sling/api/resource/path/PathSet.java

```diff
@@ -199,7 +219,7 @@ public PathSet getSubset(final PathSet set) {
      */
     @Override
     public Iterator<Path> iterator() {
-        return Collections.unmodifiableSet(this.paths).iterator();
+        return this.paths.iterator();
```

Comment:
I think this changes has the danger that the call has a way to modifiy the set (by using remove() on the iterator). So we might need to wrap the iterator into an unmodifiable variant

Reply:
fixed in 004ae77  by introducing a UnmodifiableIterator.